subsys/bluetooth/audio/pacs.c

```diff
@@ -270,36 +265,35 @@ static int set_supported_contexts(uint16_t contexts, uint16_t *supported,
 		return 0;
 	}
 
-	err = k_work_reschedule(&supported_contexts_work, PAC_NOTIFY_TIMEOUT);
-	if (err < 0) {
+	*supported = contexts;
+
+	err = supported_contexts_notify();
+	if (err) {
+		*supported = tmp_supported;
+
 		return err;
 	}
 
-	*supported = contexts;
-
 	/* Update available contexts if needed*/
 	if ((contexts & *available) != *available) {
 		*available = *available & contexts;
-		err = k_work_reschedule(&available_contexts_work,
-					PAC_NOTIFY_TIMEOUT);
-		if (err < 0) {
-			LOG_WRN("Update available contexts notify failed: %d", err);
+		err = available_contexts_notify();
```

Comment:
Since `available` shall be a subset of `supported`, if we do not revert the change to `*supported` here, we may end in a situation where `available` may no longer be a subset. 

I don't have a good solution for this, because I don't think we can stop the `supported` notification. So in this case we may need to just set `available` regardless of whether notifications fails. 

One alternative is to let the higher layer handle this, and reject requests to set supported to a value that isn't compatible with the `available` context. This approach enforces the caller to ensure that `available` is a subset of the new supported context, but may require the caller to do
```
set_available_contexts
set_supported_contexts
set_available_contexts
```